copyspecial.py

```diff
@@ -6,36 +6,86 @@
 # Google's Python Class
 # http://code.google.com/edu/languages/google-python-class/
 
+
+"""
+Assignment: CopySpecial
+
+Description: program that takes in a directory to find special files.
+Shows the absolute path to the files and then prints them to the stdout.
+Two options are allowed. --tozip and --todir.
+--todir copies special files to directory specified
+--tozip runs command to zip files and zips them there
+
+Author: Aaron Jackson
+Github: TimeApollo
+"""
+__author__ = 'TimeApollo'
+
 import sys
 import re
 import os
 import shutil
 import commands
 import argparse
 
-"""Copy Special exercise
-"""
+"""Copy Special exercise"""
+
+
+def zip_to(paths, zippath):
+    """given file paths and a zippath name, zips files"""
+
+    cmd = 'zip -j ' + zippath
+    for path in paths:
+        cmd += ' ' + path
+
+    print("Command I'm going to do: " + cmd)
+
+    (status, output) = commands.getstatusoutput(cmd)
+
+    if status:
+        print(output)
+        sys.exit(status)
+
+
+def copy_to(paths, dir):
+    """given list of paths, copies files into given directory"""
+    for path in paths:
+        shutil.copy(path, dir)
```

Comment:
This will break if the directory doesn't exist. Use
os.path.exists and os.makedirs to check if the path exists and create the directory if not.

Reply:
Updated this to check to see if the path exists and if it doesnt, it now makes the directory and adds the files there.